storage/client/src/main/java/com/azure/storage/file/FileServiceClient.java

```diff
@@ -1,52 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
 package com.azure.storage.file;
 
 import com.azure.core.http.rest.Response;
 import com.azure.core.http.rest.VoidResponse;
 import com.azure.storage.file.models.FileServiceProperties;
 import com.azure.storage.file.models.ListSharesOptions;
 import com.azure.storage.file.models.ShareItem;
-import reactor.core.publisher.Flux;
-import reactor.core.publisher.Mono;
 
 import java.util.Map;
 
-public class FileServiceClient {
-
+public final class FileServiceClient {
     private final FileServiceAsyncClient client;
 
-    FileServiceClient() {
-        throw new UnsupportedOperationException();
+    FileServiceClient(FileServiceAsyncClient client) {
+        this.client = client;
     }
 
-    public static FileServiceClientBuilder syncBuilder() {
-        throw new UnsupportedOperationException();
+    public static FileServiceClientBuilder builder() {
+        return new FileServiceClientBuilder();
     }
 
     public String url() {
-        throw new UnsupportedOperationException();
+        return client.url();
     }
 
     public ShareClient getShareClient(String shareName) {
-        throw new UnsupportedOperationException();
+        return new ShareClient(client.getShareAsyncClient(shareName));
+    }
+
+    public Iterable<ShareItem> listShares() {
+        return listShares(null);
+    }
+
+    public Iterable<ShareItem> listShares(ListSharesOptions options) {
+        return client.listShares(options).toIterable();
+    }
+
+    public Response<FileServiceProperties> getProperties() {
+        return client.getProperties().block();
     }
 
-    public Flux<ShareItem> listShares(ListSharesOptions options) {
-        throw new UnsupportedOperationException();
+    public VoidResponse setProperties(FileServiceProperties properties) {
+        return client.setProperties(properties).block();
     }
 
-    public Mono<Response<FileServiceProperties>> getProperties() {
-        throw new UnsupportedOperationException();
+    public ShareClient createShare(String shareName) {
```

Comment:
The return type is suppose tobe Response<ShareClient> based on the design. Is there any change here?

Reply:
I'll change them back to response, this was chosen to have the method more strongly correlate to it making a call to the service right?

Reply:
My understanding here is we are able to check the service response status, like 202 or 400.